localflavor/ir/forms.py

```diff
@@ -0,0 +1,90 @@
+"""Iranian-specific form helpers."""
+from __future__ import unicode_literals
+
+import re
+
+from django.core.exceptions import ValidationError
+from django.forms.fields import Field, RegexField, Select
+from django.utils.translation import ugettext_lazy as _
+
+from .ir_provinces import PROVINCE_CHOICES
+
+
+class IRProvinceSelect(Select):
+    """
+    A Select widget that uses a list of Iran provinces cities as its choices.
+
+    .. versionadded:: 2.2
+    """
+
+    def __init__(self, attrs=None):
+        super(IRProvinceSelect, self).__init__(attrs, choices=PROVINCE_CHOICES)
+
+
+class IRPostalCodeField(RegexField):
+    """
+    A form field that validates its input as an Iran postal code.
+
+    Valid form is XXXXXXXXXX where X represents integer.
+
+    Validate code:
+        - don't use 0 in first 5 digit
+        - don't use 2 in postal code
+        - First 4 digit is not the same
+        - The 5th digit cannot be 5
+        - all digits aren't the same
+
+    .. versionadded:: 2.2
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a postal code in the format XXXXXXXXXX - digits only'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(IRPostalCodeField, self).\
+            __init__(r'\b(?!(\d)\1{3})[13-9]{4}[1346-9][013-9]{5}\b$', *args, **kwargs)
+
+    def clean(self, value):
+        if value not in self.empty_values:
+            value = value.replace(' ', '')
+        return super(IRPostalCodeField, self).clean(value)
+
+
+class IRIDNumberField(Field):
+    """
+    A form field that validates its input as an Iranian identification number.
+
+    Valid form is per the Iranian ID specification.
+
+    persian documentation :
+        http://www.aliarash.com/article/codemeli/codemeli.htm
+
+        i can't find good english doc so i propose to use google translate for this one
```

Comment:
I think we can be a little more clear with this comment:
```suggestion
        There isn't good English documentation available for the Iranian identification number. Non-Persian speakers
        will need to use an online translation service to read this documentation.
```